### ISSUE TITLE
Drawing multiple elements but they only display in chunks, not the moment they are drawn - buffering problem?

### ISSUE THREAD

**Issue by SamIAm2000** [acamposuribe/p5.brush#20]
I'm trying to draw shapes on key pressed, e.g. whenever I press a key, a shape is drawn. However, I've noticed that the shapes are displayed only after several function calls and they don't display on the screen instantly. I wonder if it could be buffering problems - is there any way to clear the buffer and push all drawn shapes to screen? Or could be something else entirely, I'm not sure...

This is a simplified version of what my code looks like:
```
function setup() {
  createCanvas(600, 600, WEBGL);
  angleMode(DEGREES);
  background("#fffceb");
  brush.scaleBrushes(1.5);
}

function draw() {
  frameRate(10);
  translate(-width / 2, -height / 2);
}

function drawCircle() {
  var color = random(palette);
  brush.set(random(available_brushes), color, 5);
  brush.fill(color, 75);
  brush.circle(random(width), random(height), random(10, 300), true);
}

function keyPressed() {
  drawCircle();
}
```

All variables are defined, and it works except for the not displaying instantly part.

**Comment by acamposuribe** [comment]
Hi @SamIAm2000!
Apologies for my slow reply. Yes, indeed, it's a buffering problem.
You can force shapes to be drawn into the canvas with brush.reDraw() and brush.reBlend()
brush.reBlend() is for watercolor fills and marker lines, and brush.reDraw() is for the rest.

The info is on the docs but not very intuitive, I agree...

**Comment by SamIAm2000** [comment]
Hi @acamposuribe, thanks for your answer! 

I tried using brush.reBlend() and brush.reDraw() but I think I still have problems. brush.reDraw() works but brush.reBlend() does not seem to, as I get the same result as when I don't use it. If I draw 3 circles all with white fill, the fill does not appear instantly as the circles are drawn, but they appear only when I try to draw a new circle with a different fill color.

I tried to randomly draw white circles on the canvas (see code below), however, the circles are drawn without any fill since (I presume) the fill color has not changed. However, if I select a random new color for the fill, filled-in circles will start to appear on the canvas after a key press if the new circle fill color is different from the previous one.
```
let palette = ["#F2E2C4", "#261D11", "#A6290D"]; //white black red
let available_brushes = ["hatch_brush"];

function setup() {
  createCanvas(600, 600, WEBGL);
  angleMode(DEGREES);
  background("#ADD8E6");
  brush.scaleBrushes(1.5);
}

function draw() {
  frameRate(10);
  translate(-width / 2, -height / 2);
}

function drawCircle() {
  //var color = random(palette);
  var color = palette[2];
  brush.set(random(available_brushes), color, 5);
  brush.fill(color, 75);
  brush.circle(random(width), random(height), random(10, 300), true);
}

function keyPressed() {
  drawCircle();
  brush.reDraw();
  brush.reBlend();
}
```
Another issue I had was that nothing appears on screen if brush.reBlend() is called before brush.reDraw(). (I don't know if this was intentional behavior sorry I just thought it was rather odd)

Also thank you guys so much for creating such a cool library, I love it so much!
